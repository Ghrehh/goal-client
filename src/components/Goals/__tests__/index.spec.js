import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow'
import { Goals } from '../';

describe('Goals', () => {
  let testContext = {};

  beforeEach(() => {
    testContext = {};

    testContext.component = (
      <Goals
        goals={[{ name: 'Goal', id: 1 }]}
        loading={false}
        error={undefined}
      />
    );
  });

  describe('snapshot', () => {
    beforeEach(() => {
      const renderer = new ShallowRenderer();
      testContext.result = renderer.render(testContext.component);
    });

    it('renders the component correctly', () => {
      expect(testContext.result).toMatchSnapshot();
    });
  });
});
