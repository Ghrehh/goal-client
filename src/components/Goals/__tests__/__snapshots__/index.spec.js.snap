// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Goals snapshot renders the component correctly 1`] = `
<LoadingAndErrorHandler
  loading={false}
>
  <div
    className="goal"
  >
    <h3
      className="goalName"
    >
      Goal
    </h3>
    <CompletionWrapped
      goal={
        Object {
          "id": 1,
          "name": "Goal",
        }
      }
    />
    <DeleteWrapped
      goalId={1}
    />
    <br />
  </div>
</LoadingAndErrorHandler>
`;
